plugins/generator-1.15.2/forge-1.15.2/templates/plant.java.ftl

```diff
@@ -374,9 +374,47 @@ import net.minecraft.block.material.Material;
             </#if>
         </#if>
 
+		<#if (data.canBePlacedOn?size > 0)>
+		protected boolean canSustainBush(BlockState state) {
+			return (<#list data.canBePlacedOn as blockPlacedOn>state.getBlock() == ${blockPlacedOn}<#if blockPlacedOn?has_next>||</#if></#list>);
+		}
+
+		public boolean canPlaceBlockAt(World worldIn, BlockPos pos) {
+			BlockState soil = worldIn.getBlockState(pos.down());
+			Block block = soil.getBlock();
+			return (<#list data.canBePlacedOn as blockPlacedOn>block == ${blockPlacedOn}<#if blockPlacedOn?has_next>||</#if></#list>);
+		}
+
+		public boolean canBlockStay(World worldIn, BlockPos pos, BlockState state) {
+			if (pos.getY() >= 0 && pos.getY() < 256) {
+				BlockState blockstate = worldIn.getBlockState(pos.down());
+				Block block = blockstate.getBlock();
+				return (<#list data.canBePlacedOn as canBePlacedOn>block == ${canBePlacedOn}<#if canBePlacedOn?has_next>||</#if></#list>);
+			} else {
+				return false;
+			}
+		}
+
+		public boolean isValidGround(BlockState state, IBlockReader worldIn, BlockPos pos) {
+			Block block = state.getBlock();
+			return (<#list data.canBePlacedOn as canBePlacedOn>block == ${canBePlacedOn}<#if canBePlacedOn?has_next>||</#if></#list>);
+		}
+
+		@Override public BlockState updatePostPlacement(BlockState stateIn, Direction facing, BlockState facingState, IWorld worldIn, BlockPos currentPos, BlockPos facingPos) {

```

Comment:
This is already handled by the super classes, so it should be removed (also for 1.16)